internal/core/ebook.go

```diff
@@ -0,0 +1,252 @@
+package core
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	fp "path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/go-shiori/shiori/internal/model"
+	"github.com/pkg/errors"
+)
+
+func EbookGenerate(req ProcessRequest) (book model.Bookmark, isFatalErr bool, err error) {
```

Comment:
```suggestion
func GenerateEbook(req ProcessRequest) (book model.Bookmark, isFatalErr bool, err error) {
```

Reply:
Do we need `isFatalErr` ?

Reply:
Ideally fatal errors only happen at startup, if we are doing this per-request fatal errors make no sense, because they will stop the server. We will log the error and return an appropriate http status code to the user depending on the error. Thanks for the changes! All of that will go away with the new API.